sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_invoices_as_base64_source.py

```diff
@@ -0,0 +1,240 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_analyze_invoices_as_base64_source.py
+
+DESCRIPTION:
+    This sample demonstrates how to analyze invoices.
+
+    See fields found on a invoice here:
+    https://aka.ms/azsdk/documentintelligence/invoicefieldschema
+
+USAGE:
+    python sample_analyze_invoices_as_base64_source.py
+
+    Set the environment variables with your own values before running the sample:
+    1) DOCUMENTINTELLIGENCE_ENDPOINT - the endpoint to your Document Intelligence resource.
+    2) DOCUMENTINTELLIGENCE_API_KEY - your Document Intelligence API key.
+"""
+
+import os
+import base64
+
+
+def analyze_invoice():
+    path_to_sample_documents = os.path.abspath(
+        os.path.join(
+            os.path.abspath(__file__),
+            "..",
+            "./sample_forms/forms/sample_invoice.jpg",
+        )
+    )
+
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.documentintelligence import DocumentIntelligenceClient
+    from azure.ai.documentintelligence.models import AnalyzeResult, AnalyzeDocumentRequest
+
+    endpoint = os.environ["DOCUMENTINTELLIGENCE_ENDPOINT"]
+    key = os.environ["DOCUMENTINTELLIGENCE_API_KEY"]
+
+    document_intelligence_client = DocumentIntelligenceClient(endpoint=endpoint, credential=AzureKeyCredential(key))
+    with open(path_to_sample_documents, "rb") as f:
+        poller = document_intelligence_client.begin_analyze_document(
+            "prebuilt-invoice", AnalyzeDocumentRequest(base64_source=f.read()), locale="en-US"
```

Comment:
You're opening a JPG. A JPEG is not base64 data, so I'm confused here?

Reply:
Ok, I understand what is happening. The RestAPI needs based64 on the wire. However, the codegen is doing the base64 encoding on behalf of customers thwn they pass bytes.

The problem here, is that the parameter is infered from the RestAPI parameter, and therefore called "base64_source", giving the feeling to customer they can pass base64. But they can't, they need to pass bytes.

This parameter needs a rename (clientName), so customers understand they need to pass raw bytes.

This should be clarified in the sample.

CC @johanste

Reply:
Yes, it's not expected to pass base64 encoded content from customers, our SDK would encode the data automatically. I have requested other languages to confirm if they have the same behavior, and requested a name change in the same email thread.